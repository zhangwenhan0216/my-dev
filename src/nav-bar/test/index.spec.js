import NavBar from '..';
import { mount, mockGetBoundingClientRect } from '../../../test';

test('render left & right slot', () => {
  const wrapper = mount(NavBar, {
    scopedSlots: {
      left: () => 'Custom Left',
      right: () => 'Custom Right',
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('render title slot', () => {
  const wrapper = mount(NavBar, {
    scopedSlots: {
      title: () => 'Custom Title',
    },
  });

  expect(wrapper).toMatchSnapshot();
});

test('placeholder prop', () => {
  const restore = mockGetBoundingClientRect({ height: 50 });

  const wrapper = mount(NavBar, {
    propsData: {
      fixed: true,
      placeholder: true,
    },
  });

  expect(wrapper).toMatchSnapshot();

  restore();
});

test('click-left event', () => {
  const wrapper = mount(NavBar, {
    propsData: {
      leftText: 'left',
    },
  });

  wrapper.find('.van-nav-bar__left').trigger('click');
  expect(wrapper.emitted('click-left')).toBeTruthy();
});

test('click-right event', () => {
  const wrapper = mount(NavBar, {
    propsData: {
      rightText: 'right',
    },
  });

  wrapper.find('.van-nav-bar__right').trigger('click');
  expect(wrapper.emitted('click-right')).toBeTruthy();
});
