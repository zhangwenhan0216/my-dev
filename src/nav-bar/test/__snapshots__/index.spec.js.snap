// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`placeholder prop 1`] = `
<div class="van-nav-bar__placeholder" style="height: 50px;">
  <div class="van-nav-bar van-nav-bar--fixed van-hairline--bottom">
    <div class="van-nav-bar__left"></div>
    <div class="van-nav-bar__title van-ellipsis"></div>
    <div class="van-nav-bar__right"></div>
  </div>
</div>
`;

exports[`render left & right slot 1`] = `
<div class="van-nav-bar van-hairline--bottom">
  <div class="van-nav-bar__left">Custom Left</div>
  <div class="van-nav-bar__title van-ellipsis"></div>
  <div class="van-nav-bar__right">Custom Right</div>
</div>
`;

exports[`render title slot 1`] = `
<div class="van-nav-bar van-hairline--bottom">
  <div class="van-nav-bar__left"></div>
  <div class="van-nav-bar__title van-ellipsis">Custom Title</div>
  <div class="van-nav-bar__right"></div>
</div>
`;
